crates/biome_analyze/src/rule.rs

```diff
@@ -513,6 +513,19 @@ impl Display for RuleDomain {
     }
 }
 
+impl Ord for RuleDomain {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // Rule domains should be in alphabetical order
+        format!("{self:?}").cmp(&format!("{other:?}"))
```

Comment:
This had the disadvantage of allocating two strings. Could we reorder the enum member instead of writing our own impls?

Reply:
It's just codegen 😅

Reply:
Oh? It still makes sense to reduces codegen execution time :)

Reply:
Though it is used only in codegen for now, it is placed in the core of the linter system. I agree to remove and use the original order from the variants. Btw, does alphabetical order fit in this case? We'll lost oppotunity to colocate domains, e.g. React and Next.js

Reply:
The order is only sued for consistent printing? In that case I think an alphabetical sorting is a good option because we already use this kind of ordering in other places.

Reply:
Yes, sorting is just for docs

Reply:
AFAIK we can't enforce alphabetical order for enum variants automatically. Manual sorting can be broken somewhen. To keep the linter core clean and enforce alphabetical order automatically, it makes sense for me to use `Debug` to sort variants like the original implementation (https://github.com/biomejs/website/pull/1797/files#diff-cb306886ea92d37b5ce007cc39b3fcf26079234cd572cc7502e05696b2d75346R67-R69).

Reply:
In that case I could use manual ordering and add a comment. However, I think we can keep the actual impl. That's fine. Thanks for the quick responses ;)